tests/production_pause_max_rev_blks_test.py

```diff
@@ -0,0 +1,177 @@
+    assert transId is not None, "setfinalizers failed"
+    assert cluster.biosNode.waitForTransFinalization(transId), f"setfinalizers transaction {transId} was not rolled into a LIB block"
+    assert cluster.biosNode.waitForLibToAdvance(), "LIB did not advance after setFinalizers"
+
+    # biosNode no longer needed
+    cluster.biosNode.kill(signal.SIGTERM)
+    cluster.waitOnClusterSync(blockAdvancing=5)
+
+    Print("Wait for LIB on all producing nodes to advance")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert producerbNode.waitForLibToAdvance(), "producerbNode did not advance LIB"
+
+    Print("Restart node0 and producerbNode with max-reversible-blocks")
+    node0.kill(signal.SIGTERM)
+    assert not node0.verifyAlive(), "node0 did not shutdown"
+    producerbNode.kill(signal.SIGTERM)
+    assert not producerbNode.verifyAlive(), "producerbNode did not shutdown"
+    maxReversibleBlocks=6
+    addSwapFlags={"--max-reversible-blocks": str(maxReversibleBlocks)}
+    node0.relaunch(chainArg="--enable-stale-production", addSwapFlags=addSwapFlags)
+    producerbNode.relaunch(chainArg="--enable-stale-production", addSwapFlags=addSwapFlags)
+
+    ####################### test 1 ######################
+    # production paused due to max reversible blocks exceeded
+
+    Print("Shutdown finalizerbNode")
+    finalizerbNode.kill(signal.SIGTERM)
+    assert not finalizerbNode.verifyAlive(), "finalizerbNode did not shutdown"
+
+    # Verify LIB stalled on node0 and producerbNode due to finalizerNode was shutdown
+    Print("Verify LIB stalled after shutdown of finalizerbNode")
+    if producerbNode.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not producerbNode.waitForLibToAdvance(timeout=5), "LIB should not advance on producerbNode after finalizerbNode was shutdown"
+    if node0.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not node0.waitForLibToAdvance(timeout=5), "LIB should not advance on node0 after finalizerbNode was shutdown"
+
+    # Wait until enough reversible blocks are produced
+    node0.getInfo()
+    node0.waitForBlock(node0.lastRetrievedLIB + maxReversibleBlocks)
+
+    # Verify node0 and producerbNode then paused due to max reversible blocks exceeded
+    Print("Verify production paused after LIB stalled")
+    assert producerbNode.missedNextProductionRound(), "producerbNode still producing after finalizerbNode was shutdown"
+    assert node0.missedNextProductionRound(), "node0 still producing after finalizerbNode was shutdown"
+
+    ####################### test 2 ###########################
+    # nodeos can restart with a higher --max-reversible-blocks
+
+    node0.getInfo()
+    prevHeadBlockNum=node0.lastRetrievedHeadBlockNum
+
+    Print("Shutdown node0")
+    node0.kill(signal.SIGTERM)
+    assert not node0.verifyAlive(), "node0 did not shutdown"
+
+    higherMaxReversibleBlocks=maxReversibleBlocks+12
```

Comment:
Add a comment that `12` is just a value chosen for test. Might be better to use 10 so no confusion that it is related to blocks per round.